scripts/lib/setup_venv.py

```diff
@@ -244,7 +244,14 @@ def do_setup_virtualenv(venv_path, requirements_file, virtualenv_args):
     activate_this = os.path.join(venv_path, "bin", "activate_this.py")
     exec(open(activate_this).read(), {}, dict(__file__=activate_this))
 
-    run(["pip", "install", "-U", "setuptools==35.0.2"])
-    run(["pip", "install", "--upgrade", "pip", "wheel"])
-    run(["pip", "install", "--no-deps", "--requirement", requirements_file])
+    try:
+        run(["pip", "install", "-U", "setuptools==35.0.2"])
+        run(["pip", "install", "--upgrade", "pip", "wheel"])
+        run(["pip", "install", "--no-deps", "--requirement", requirements_file])
+    except subprocess.CalledProcessError:
+        # Might be a failure due to network connection issues. Retrying...
+        print(WARNING + "`pip install` failed; retrying..." + ENDC)
+        run(["pip", "install", "-U", "setuptools==35.0.2"])
+        run(["pip", "install", "--upgrade", "pip", "wheel"])
+        run(["pip", "install", "--no-deps", "--requirement", requirements_file])
```

Comment:
I don't like the code duplication here (it will likely lead to pain later); we should extract an `install_venv_deps` function and just call it both in the try and except blocks.